test/amiadapters/test_sentryx.py

```diff
@@ -0,0 +1,279 @@
+        self.assertEqual("1", meter.account_id)
+        self.assertEqual(654419700, meter.device_id)
+        self.assertEqual("OK", meter.device_status)
+        self.assertEqual("NotRDM", meter.service_status)
+        self.assertEqual("10327 SW SHAD RD, ", meter.street)
+        self.assertEqual("CROOKED RIVER RANCH                               ", meter.city)
+        self.assertEqual(None, meter.state)
+        self.assertEqual("97760", meter.zip)
+        self.assertEqual("Mueller Systems SSR Ext-3/4(CF)-Pit  Plastic   Positive Displacement ", meter.description)
+        self.assertEqual(None, meter.manufacturer)
+        self.assertEqual("", meter.install_notes)
+        self.assertEqual("2022-02-08T20:16:41", meter.install_date)
+        self.assertEqual("3/4\"", meter.meter_size)
+
+        calls = [
+            mock.call('https://api.sentryx.io/v1-wm/sites/crookedriverranchor/devices', headers={'Authorization': 'key'}, params={'pager.skip': 0, 'pager.take': 25}),
+            mock.call('https://api.sentryx.io/v1-wm/sites/crookedriverranchor/devices', headers={'Authorization': 'key'}, params={'pager.skip': 1, 'pager.take': 25})
+        ]
+        self.assertListEqual(calls, mock_get.call_args_list)
+    
+    @mock.patch('requests.get', side_effect=[mocked_get_devices_response_first_page(), mocked_response_500()])
+    def test_extract_all_meters__non_200_status_code(self, mock_get):
+        config = AMIAdapterConfiguration(
+            output_folder="output",
+            sentryx_api_key="key",
+        )
+        adapter = SentryxAdapter(config)
+        result = adapter._extract_all_meters()
+        self.assertEqual(0, len(result))
+    
+    @mock.patch('requests.get', side_effect=[mocked_get_consumption_response_first_page(), mocked_get_consumption_response_last_page()])
+    def test_extract_consumption_for_all_meters(self, mock_get):
+        config = AMIAdapterConfiguration(
+            output_folder="output",
+            sentryx_api_key="key",
+        )
+        adapter = SentryxAdapter(config)
+        result = adapter._extract_consumption_for_all_meters()
+        expected = [
+            SentryxMeterWithReads(device_id=1, units='CF', data=[SentryxMeterRead(time_stamp='2024-07-07T01:00:00', reading=116233.61)]), 
+            SentryxMeterWithReads(device_id=2, units='CF', data=[SentryxMeterRead(time_stamp='2024-07-08T01:00:00', reading=22.61)])
+        ]
+        self.assertListEqual(expected, result)
+
+        calls = [
+            mock.call('https://api.sentryx.io/v1-wm/sites/crookedriverranchor/devices/consumption', headers={'Authorization': 'key'}, params={'skip': 0, 'take': 25, 'StartDate': '2025-03-09T11:02:26.011959', 'EndDate': '2025-03-11T11:02:26.011959'}),
+            mock.call('https://api.sentryx.io/v1-wm/sites/crookedriverranchor/devices/consumption', headers={'Authorization': 'key'}, params={'skip': 2, 'take': 25, 'StartDate': '2025-03-09T11:02:26.011959', 'EndDate': '2025-03-11T11:02:26.011959'})
```

Comment:
@mdowell12 same maybe goes for utility names, for example in API endpoints if we can abstract that away or use a dummy name.